src/addrman.cpp

```diff
@@ -1065,6 +1099,15 @@ int AddrManImpl::CheckAddrman() const
     if (nKey.IsNull())
         return -16;
 
+    int cnt_new{0};
+    int cnt_tried{0};
+    for (auto net : map_counts) {
+        cnt_new += net.second.first;
+        cnt_tried += net.second.second;
+    }
+    if (cnt_new != nNew || cnt_tried != nTried) {
```

Comment:
Given this dependency, should we drop `nNew` and `nTried` altogether?

Reply:
I think that's not so streighforward - `nNew` and `nTried` are part of the serialization after all. Also, I don't like that then we'd need to build the sum over all networks every time we needed `nNew` or `nTried` (as the check code does).

Reply:
>Also, I don't like that then we'd need to build the sum over all networks every time we needed nNew or nTried (as the check code does).

Well, I don't think something being inside the `Check` is a good justification — it's the other way around, `Check` is needed to make sure those values are sane. Furthermore, the sum over all networks is 5 addition operations, right? 

However, I agree that dropping these values might be not trivial due to serialization. Not sure I'll find time to prototype that, but maybe someone will :) Otherwise feel free to close this suggestion.

Reply:
Also, see [this](https://github.com/bitcoin/bitcoin/pull/26847#discussion_r1065149504) :)

Reply:
I've changed the check such that it no longer compares with `nNew` and `nTried` but does a recount, see below for more context (of course the relation is still there...).